x/concentrated-liquidity/swaps.go

```diff
@@ -414,6 +414,11 @@ func (k Keeper) computeOutAmtGivenIn(
 		}
 	}
 
+	// Note, this should be impossible to reach but we leave it as a defense-in-depth measure.
+	if swapState.amountSpecifiedRemaining.IsNegative() {
+		return sdk.Coin{}, sdk.Coin{}, PoolUpdates{}, sdk.Dec{}, fmt.Errorf("over charge problem swap out given in by (%s)", swapState.amountSpecifiedRemaining)
+	}
+
```

Comment:
```suggestion
	// Note, this should be impossible to reach but we leave it as a defense-in-depth measure.
	if swapState.amountSpecifiedRemaining.IsNegative() {
		return sdk.Coin{}, sdk.Coin{}, PoolUpdates{}, sdk.Dec{}, fmt.Errorf("over charge problem swap out given in by (%s)", swapState.amountSpecifiedRemaining)
	}

```

I reproduced this case with swap out given in where amount in consumed ends up being greater than amountInRemaining due to rounding up. The test is at the swap strategy level of abstraction here:
https://github.com/osmosis-labs/osmosis/blob/235157772c624f255869c59b72a46e8379b5d802/x/concentrated-liquidity/swapstrategy/zero_for_one_test.go#L178-L199

This results in negative amountInRemaining [here](https://github.com/osmosis-labs/osmosis/blob/55126a8e75417222b067eaa65f04b13d83826c7f/x/concentrated-liquidity/swaps.go#L381). However, contrary to in given out, I believe that it is acceptable.

The reason is that we end up subtracting the final negative amountInRemaining [here](https://github.com/osmosis-labs/osmosis/blob/55126a8e75417222b067eaa65f04b13d83826c7f/x/concentrated-liquidity/swaps.go#L419) and then rounding up. As a result, worst case, we upcharge the user by 1 base unit.

Therefore, I think we should not have this check


Any thoughts?

Reply:
In the world of "normal" swaps, how often will a user run into this? If the answer is <0.01 percent of the time, I saw we just keep the guardrail in place. Accepting negative numbers doesn't seem ideal. If we decide it is okay, we should cap the negative number to what it is we expect, so we don't accidentally overcharge the user by more than 1 ULP

Reply:
I don't have a specific estimate but I would expect this to be quite rare.

Additionally, none of the tests in our suite, including randomized, have run into this. So keeping this to prevent accidental overcharge might make sense.

Reply:
Im for the "diminished UX" in place of safety, especially when this will rarely happen

Reply:
Sound great, thanks. Agreed.

I'm going to merge this but also ping @AlpinYukseloglu and @ValarDragon to share opinions